plenum/test/node_catchup/test_node_catchup_with_connection_problem.py

```diff
@@ -0,0 +1,186 @@
+                                                    tdir,
+                                                    allPluginsPath,
+                                                    monkeypatch,
+                                                    lost_count):
+    node_to_disconnect = txnPoolNodeSet[-1]
+
+    def unpatch_after_call(proof, frm):
+        global call_count
+        call_count += 1
+        if call_count >= lost_count:
+            # unpatch processLedgerStatus after lost_count calls
+            monkeypatch.undo()
+            call_count = 0
+
+    sdk_send_random_and_check(looper, txnPoolNodeSet,
+                              sdk_pool_handle, sdk_wallet_steward, 5)
+
+    # restart node
+    disconnect_node_and_ensure_disconnected(looper,
+                                            txnPoolNodeSet,
+                                            node_to_disconnect)
+    looper.removeProdable(name=node_to_disconnect.name)
+    sdk_send_random_and_check(looper, txnPoolNodeSet,
+                              sdk_pool_handle, sdk_wallet_steward,
+                              2)
+
+    nodeHa, nodeCHa = HA(*node_to_disconnect.nodestack.ha), HA(
+        *node_to_disconnect.clientstack.ha)
+    config_helper = PNodeConfigHelper(node_to_disconnect.name, tconf,
+                                      chroot=tdir)
+    node_to_disconnect = TestNode(node_to_disconnect.name,
+                                  config_helper=config_helper,
+                                  config=tconf,
+                                  ha=nodeHa, cliha=nodeCHa,
+                                  pluginPaths=allPluginsPath)
+    # patch processLedgerStatus
+    monkeypatch.setattr(node_to_disconnect.ledgerManager,
+                        'processConsistencyProof',
+                        unpatch_after_call)
+    # add node_to_disconnect to pool
+    looper.add(node_to_disconnect)
+    txnPoolNodeSet[-1] = node_to_disconnect
+    looper.run(checkNodesConnected(txnPoolNodeSet))
+    waitNodeDataEquality(looper, node_to_disconnect, *txnPoolNodeSet)
+
+
+def test_catchup_with_lost_last_consistency_proof(txnPoolNodeSet,
+                                                  looper,
+                                                  sdk_pool_handle,
+                                                  sdk_wallet_steward,
+                                                  tconf,
+                                                  tdir,
+                                                  allPluginsPath,
+                                                  monkeypatch,
+                                                  lost_count):
+    node_to_disconnect = txnPoolNodeSet[-1]
+    tmp_method = node_to_disconnect.ledgerManager.canProcessConsistencyProof
+
+    def unpatch_after_call(proof):
+        if node_to_disconnect.ledgerManager.spylog.count(
+                LedgerManager.processConsistencyProof) <= 2:
+            tmp_method(proof)
```

Comment:
But `tmp_method` is for the old instance of disconnected node (before restart)

Reply:
Fixed.